File: Aman_Bhaiya_DSA_List/Strings/LQ42.java
package Aman_Bhaiya_DSA_List.Strings;

import java.util.Stack;

public class LQ42 {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(c == '(' || c == '{' || c == '[')
                st.push(c);
                
            else{
                if(c == ')'){
                    if(st.empty() || st.pop() != '(')
                        return false;
                }
                if(c == '}'){
                    if(st.empty() || st.pop() != '{')
                        return false;
                }
                if(c == ']'){
                    if(st.empty() || st.pop() != '[')
                        return false;
                }
            }
        }
        
        return st.empty();
    }
}

File: Aman_Bhaiya_DSA_List/Strings/LQ42.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if i == '(' or i == '{' or i == '[':
                stack.append(i)
            else:
                if i == ')':
                    if len(stack) == 0 or stack.pop() != '(':
                        return False
                if i == '}':
                    if len(stack) == 0 or stack.pop() != '{':
                        return False
                if i == ']':
                    if len(stack) == 0 or stack.pop() != '[':
                        return False
        if len(stack) == 0:
            return True
        else:
            return False